core/src/bundle_stage.rs

```diff
@@ -122,6 +123,53 @@ struct AllExecutionResults {
     pub post_balances: (TransactionBalances, TransactionTokenBalances),
 }
 
+struct BundleReservedSpace {
+    current_tx_block_limit: u64,
+    current_bundle_block_limit: u64,
+    initial_allocated_cost: u64,
+}
+
+impl BundleReservedSpace {
+    fn reset_reserved_cost(&mut self, working_bank: &Arc<Bank>) {
+        self.current_bundle_block_limit = MAX_BLOCK_UNITS;
+        self.current_tx_block_limit = MAX_BLOCK_UNITS - self.initial_allocated_cost;
+
+        working_bank.write_cost_tracker().unwrap().set_limits(
+            MAX_WRITABLE_ACCOUNT_UNITS,
+            self.current_tx_block_limit,
+            MAX_VOTE_UNITS,
+        );
+
+        info!(
+            "Slot: {}. Cost Limits Reset. Bundle: {}, TX: {}",
+            working_bank.slot(),
+            self.current_bundle_block_limit,
+            self.current_tx_block_limit,
+        );
+    }
+
+    fn bundle_block_limit(&self) -> u64 {
+        self.current_bundle_block_limit
+    }
+
+    fn tx_block_limit(&self) -> u64 {
+        self.current_tx_block_limit
+    }
+
+    fn update_reserved_cost(&mut self, working_bank: &Arc<Bank>) {
+        if self.current_tx_block_limit != self.current_bundle_block_limit
+            && working_bank.tick_height() > working_bank.ticks_per_slot() * 4 / 5
+        {
+            info!(
```

Comment:
we missing call to set_limits here?

Reply:
oh nvmd, can you add comment in here on why we don't do it? seems like we need to "atomically swap" compute units or something

Reply:
i think it does need to be called here, else it wouldn't actually go into effect until a bundle was processed

Reply:
it seems like it’d be applied later on slot boundary check with working bank?

Reply:
oh nvmd i think we need it here. so in that case, for last 70-80ms of block, compute limit back to 48m units. if we get bundle in that time, maybe fits maybe doesn't

Reply:
right, and without this call, if we don't get a bundle, then it never resets to 48M